src/riak_ql_ddl.erl

```diff
@@ -379,16 +382,21 @@ is_query_valid_result(Res) ->
 
 -spec are_filters_valid(module(), [filter()]) -> true | {false, [query_syntax_error()]}.
 are_filters_valid(Mod, Where) ->
-    Errors = fold_where_tree(fun(_, Clause, Acc) ->
-                                     is_filters_field_valid(Mod, Clause, Acc)
-                             end, [], Where),
+    {_, Errors} = mapfold_where_tree(
+        fun (_, Op, Acc_x) when Op == and_; Op == or_ ->
+                {ok, Acc_x};
+            (_, Clause, Acc) ->
+                {Clause, is_filters_field_valid(Mod, Clause, Acc)}
+        end, [], Where),
     case Errors of
         [] -> true;
         _  -> {false, Errors}
     end.
 
-%% the terminal case of "a = 2"
-is_filters_field_valid(Mod, {Op, Field, {RHS_type, RHS_Val}}, Acc1) ->
+is_filters_field_valid(_, {Op, {identifier,FieldName}, _}, Acc1) when (Op == '/' orelse Op == '+' orelse Op == '-' orelse Op == '*') ->
+    [{arithmetic_on_identifier, FieldName} | Acc1];
```

Comment:
It seems that there should be a call to `Mod:is_field_valid/1` for the identifier when it is w/i an arithmetic operation as well.